test/specs/change-tabs-sometimes-match.js

```diff
@@ -0,0 +1,41 @@
+const {
+  fixture,
+  matchesKeyboard,
+  matchesMouse,
+  FOCUS_RING_STYLE
+} = require('./helpers');
+const { Key, By } = require('selenium-webdriver');
+const expect = require('expect');
+const driver = global.__driver;
+
+describe('change tabs, only match if elements had focus-visible', function() {
+  beforeEach(function() {
+    // Note: For focus to enter the page properly with this fixture I had
+    // to make sure the div had some width/height.
+    // This seems like a geckodriver bug.
+    return fixture('change-tabs-sometimes-match.html');
+  });
+
+  it('should retain .focus-visible if the user switches tabs and an element had .focus-visible from keyboard', async function() {
+    let body = await driver.findElement(By.css('body'));
+    let el = await driver.findElement(By.css('#el'));
+    await body.click();
+    await body.sendKeys(Key.TAB);
+    await el.sendKeys(Key.SPACE);
+    await driver.sleep(4000); // sleep while we open and close a new tab.
```

Comment:
Any chance we could wait for the visibility events to fire instead of this terrifying timeout?

Reply:
oh this may actually be possible, I didn't know there was an executeAsyncScript command.

Reply:
http://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_WebDriver.html

Reply:
I tried mucking around with executeAsyncScript but it seems like it was blocking the next step in the driver instructions that actually opens the new window. I'm not in love with the `sleep()` either, Selenium is kinda awful. I'd like to just move on. I've run the test 3 times through travis and it works. I won't be surprised if it becomes flakey someday. At which point I made just change it to a `.skip()` and only run it locally as a sanity check or invest more time in debugging Selenium.